File: cc/001.cc
class Solution {
public:
    bool Find(int target, vector<vector<int> > array) {
        int row = array.size() - 1, col = 0;
        while (true)
        {
            if (col == array[0].size() || row == -1)
                return false;
            if (target == array[row][col])
                return true;
            else if (target > array[row][col])
                col++;
            else
                row--;
        }
    }
};

File: python/001.py
# -*- coding: utf-8 -*-
# /usr/bin/python3
# Creator: Hemajun
# Description: 001
class Solution:
    # array 二维列表
    def Find(self, target, array):
        row = len(array)
        if row > 0:
            col = len(array[0])
            i = row - 1
            j = 0
            while True:
                if j == col or i < 0:
                    return False
                elif target == array[i][j]:
                    return True
                elif target > array[i][j]:
                    j += 1
                else:
                    i -= 1
        return False